cpp/oneapi/dal/backend/primitives/reduction/test/reduction_1d_dpc.cpp

```diff
@@ -103,4 +154,26 @@ TEMPLATE_LIST_TEST_M(reduction_test_random_1d,
     this->test_1d_reduce();
 }
 
+TEMPLATE_LIST_TEST_M(infinite_sum_test_random_1d,
+                     "Randomly filled array with infinite sum",
+                     "[reduction][1d][small]",
+                     finiteness_types) {
+    SKIP_IF(this->not_float64_friendly());
+    this->generate(true);
```

Comment:
you can simplify this test via
const bool use_infnan = GENERATE(0, 1);
this->generate(use_infnan);
this->test_1d_reduce();
and thats it(instead of 4 lines)